src/psn/mod.rs

```diff
@@ -141,6 +169,50 @@ impl UpdateInfo {
 
         Ok(info)
     }
+
+    pub async fn merge_parts(&self, tx: Sender<MergeStatus>, download_path: &PathBuf) -> Result<(), MergeError> {
+        if !self.packages.iter().all(|pkg| pkg.part_number.is_some()) {
+            return Err(MergeError::PackagesUnmergable(String::from("some packages for the update are not a partial packages")));
+        }
+
+        let mut packages_sorted_by_part_number = self.packages.clone();
+        packages_sorted_by_part_number.sort_by_key(|pkg| pkg.part_number.unwrap());
+        let package_download_path = create_new_pkg_path(&download_path, &self.title_id, &self.title());
+
+        info!("Starting merge for {}", self.title());
+
+        for package in self.packages.iter() {
+            let file_name = match package.file_name() {
+                Some(name) => name,
+                None => return Err(MergeError::FilepathMismatch(String::from("could not deduce filename from a package url")))
+            };
+
+            let part_number = package.part_number.unwrap();
+            let expected_end_of_file_name = format!("_{}.pkg", part_number - 1);
+            if !file_name.ends_with(&expected_end_of_file_name) {
+                return Err(MergeError::FilepathMismatch(String::from("package name does not end with expected index and extension")))
+            }
+
+            let merged_file_name = file_name.replace(&expected_end_of_file_name, ".pkg");
+            let mut merged_path = package_download_path.clone();
+            merged_path.push(&merged_file_name);
+            let mut package_path = package_download_path.clone();
+            package_path.push(&file_name);
+            match copy_pkg_file( &package_path, &merged_path, package.offset).await {
```

Comment:
There's an extra space here that's tickling my OCD
```suggestion
            match copy_pkg_file(&package_path, &merged_path, package.offset).await {
```

Reply:
Done